### PR TITLE
Updates to the Programmer guide section of the manual

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request integrates some discussions and explanations of core algorithms that are adapted from the new LAMMPS paper.

**Author(s)**

Axel Kohlmeyer, Temple U, plus the authors of the LAMMPS paper.

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system

### CODE REVIEW

**Comment by akohlmey** [comment]
@sjplimp @athomps @stanmoore1 @rbberger
Since the new LAMMPS paper is now accepted, has been assigned a DOI, and the "pre-proof version" has been posted online at https://doi.org/10.1016/j.cpc.2021.108171, we should consider including this pull request into the stable version and also update the information about citing LAMMPS accordingly. Even though we cannot provide the exact reference until later, we can include the DOI link which will point to the correct citation as soon as it is available and we can update the text in the manual and the website then as well. Since a large number of people will be using the stable version, I think it would be beneficial to get this included.
At the very least the reference to the new paper.

**Comment by athomps** [comment]
Sounds good to me.

**Comment by sjplimp** [comment]
Including DOI is fine.  As I mentioned before, if this content is going into the programmer guide, it needs to clearly say this content, esp the figures, is from the paper.

**Comment by akohlmey** [comment]
> Including DOI is fine. As I mentioned before, if this content is going into the programmer guide, it needs to clearly say this content, esp the figures, is from the paper.

Made some changes to do that. Please check it out. If it is not clear enough, please suggest how to improve.

File: doc/src/Intro_citing.rst [hunk 1]
```diff
@@ -4,28 +4,40 @@ Citing LAMMPS
...
+the neighbor finding, basic communications algorithms, and how users and
+developers have contributed to LAMMPS is:
+
+  `LAMMPS - A flexible simulation tool for particle-based materials modeling at the atomic, meso, and continuum scales, Comp. Phys. Comm. (accepted 09/2021), DOI:10.1016/j.cpc.2021.108171 <https://doi.org/10.1016/j.cpc.2021.108171>`_
+
+So any project using LAMMPS or a derivative application that uses LAMMPS
+as a simulation engine should cite this paper.  The paper is expected to
+be published in its final form under the same DOI in the first half of 2022.
+
+The original publication describing LAMMPS in its early implementations is:
```
Comment (sjplimp) [inline doc/src/Intro_citing.rst:27]: The original pub describing the parallel algorithms used in the initial versions of LAMMPS is:

File: doc/src/Intro_citing.rst [hunk 2]
```diff
@@ -4,28 +4,40 @@ Citing LAMMPS
...
 ^^^^^^^^^^^^^^^
 
-Since LAMMPS is a community project, there is not a single one
-publication or reference that describes **all** of LAMMPS.
-The canonical publication that describes the foundation, that is
-the basic spatial decomposition approach, the neighbor finding,
-and basic communications algorithms used in LAMMPS is:
+Since LAMMPS is a community project there is not one single publication
+or reference that describes **all** of LAMMPS.  There are publications
+describing the core of LAMMPS, but then also publications describing the
```
Comment (sjplimp) [inline doc/src/Intro_citing.rst:9]: but also pubs describing particular models or algorithms implemented in LAMMPS

File: doc/src/Intro_citing.rst [hunk 3]
```diff
@@ -4,28 +4,40 @@ Citing LAMMPS
...
-describing the developments and improvements of LAMMPS in the 25 years
-since then is currently in preparation.
+The canonical publication that describes the basic features, the source
+code design, the program structure, the spatial decomposition approach,
+the neighbor finding, basic communications algorithms, and how users and
+developers have contributed to LAMMPS is:
+
+  `LAMMPS - A flexible simulation tool for particle-based materials modeling at the atomic, meso, and continuum scales, Comp. Phys. Comm. (accepted 09/2021), DOI:10.1016/j.cpc.2021.108171 <https://doi.org/10.1016/j.cpc.2021.108171>`_
+
+So any project using LAMMPS or a derivative application that uses LAMMPS
```
Comment (sjplimp) [inline doc/src/Intro_citing.rst:23]: So a project ...

File: doc/src/Intro_citing.rst [hunk 4]
```diff
@@ -4,28 +4,40 @@ Citing LAMMPS
...
+LAMMPS.
 
- `S. Plimpton, Fast Parallel Algorithms for Short-Range Molecular Dynamics, J Comp Phys, 117, 1-19 (1995). <http://www.sandia.gov/~sjplimp/papers/jcompphys95.pdf>`_
+.. _lammps_paper:
 
-So any project using LAMMPS (or a derivative application using LAMMPS as
-a simulation engine) should cite this paper. A new publication
-describing the developments and improvements of LAMMPS in the 25 years
-since then is currently in preparation.
+The canonical publication that describes the basic features, the source
```
Comment (sjplimp) [inline doc/src/Intro_citing.rst:16]: The latest canonical ...

File: doc/src/Developer_parallel.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
+Parallel algorithms
+-------------------
+
+LAMMPS is from ground up designed to be running in parallel using the
```
Comment (sjplimp) [inline doc/src/Developer_parallel.rst:4]: LAMMPS is designed to enable running in parallel ...

File: doc/src/Developer_parallel.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+MPI standard with distributed data via domain decomposition.  The
+parallelization has to be efficient to enable good strong scaling (=
```
Comment (sjplimp) [inline doc/src/Developer_parallel.rst:6]: ... parallelization aims to be efficient so as to enable ...

File: doc/src/Developer_parallel.rst [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+good speedup for the same system) and good weak scaling (= the
+computational cost of enlarging the system is proportional to the system
+size).  Additional parallelization using GPUs or OpenMP can then be
```
Comment (sjplimp) [inline doc/src/Developer_parallel.rst:9]: can also be applied

File: doc/src/Developer_parallel.rst [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+applied within the sub-domain assigned to an MPI process.  For clarity,
+most of the following illustrations show the 2d simulation case. The
+underlying algorithms in those cases, however, apply to both 2d and 3d
+cases equally well.
+
+.. note::
+
+   The text and most of the figures in this chapter were adapted
```
Comment (sjplimp) [inline doc/src/Developer_parallel.rst:17]: thanks

**Review by sjplimp — CHANGES_REQUESTED** [review]

**Review by athomps — APPROVED** [review]
I approve

**Comment by akohlmey** [comment]
@sjplimp it looks like we also need to update the links to the original JCP paper on your home page. When I try to follow them I get an error about "too many redirects".

**Review by sjplimp — APPROVED** [review]

**Pull request merged**